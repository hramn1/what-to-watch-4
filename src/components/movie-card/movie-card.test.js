import React from "react";
import renderer from "react-test-renderer";
import FilmCard from './movie-card.jsx';

const films = [
  {
    id: `1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: `2`,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
  },
  {
    id: `3`,
    title: `Macbeth`,
    poster: `img/macbeth.jpg`,
  },
  {
    id: `4`,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
  },
  {
    id: `5`,
    title: `We need to talk about Kevin`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: `6`,
    title: `Snatch`,
    poster: `img/snatch.jpg`,
  },
  {
    id: `7`,
    title: `Revenant`,
    poster: `img/revenant.jpg`,
  },
  {
    id: `8`,
    title: `Johnny English`,
    poster: `img/johnny-english.jpg`,
  },
];
const isPlaying = true;
const muted = `muted`;
for (let item of films) {
  it(`FilmCard rendering`, () => {
    const tree = renderer
  .create(<FilmCard
    film = {item}
    title = {item.title}
    img = {item.poster}
    isPlaying = {isPlaying}
    muted = {muted}
    onMouseHover = {() =>{}}
    onFilmCardClick = {()=>{}}
    onIsPlayingChange = {()=>{}}
  />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
}
