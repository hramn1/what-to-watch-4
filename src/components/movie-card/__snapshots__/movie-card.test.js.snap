// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard rendering 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 2`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/bohemian-rhapsody.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 3`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/macbeth.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Macbeth
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 4`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/aviator.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Aviator
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 5`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/we-need-to-talk-about-kevin.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      We need to talk about Kevin
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 6`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/snatch.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Snatch
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 7`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/revenant.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Revenant
    </a>
  </h3>
</article>
`;

exports[`FilmCard rendering 8`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/johnny-english.jpg"
      width="280"
    >
      <source
        type="video/webm; codecs=\\"vp8, vorbis\\""
      />
      <source
        type="video/ogg; codecs=\\"theora, vorbis\\""
      />
      <source
        type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Johnny English
    </a>
  </h3>
</article>
`;
